examples/with-next/pages/index.tsx

```diff
@@ -0,0 +1,249 @@
+            </button>
+          )}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+interface SearchCardProps {
+  onSuccess(offer: Offer): void
+}
+
+const SearchCard: FC<SearchCardProps> = ({ onSuccess }) => {
+  const [sort, setSort] = useState<'total_amount' | 'total_duration'>(
+    'total_amount'
+  )
+  const [origin, setOrigin] = useState('JFK')
+  const [destination, setDestination] = useState('LHR')
+  const [isFetching, setIsFetching] = useState(false)
+  const [_errors, setErrors] = useState<ApiResponseError[]>([])
+
+  const fetchCheapestOffer = async () => {
+    setIsFetching(true)
+    const { data, errors } = await (
+      await fetch('/api/search', {
+        method: 'post',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          origin,
+          destination,
+          sort,
+        }),
+      })
+    ).json()
+
+    if (data) {
+      onSuccess(data)
+    } else {
+      setErrors(errors)
+      // eslint-disable-next-line
+      console.log(errors)
+    }
+
+    setIsFetching(false)
+  }
+
+  return (
+    <Fragment>
+      <h4>Example booking flow</h4>
+      <h2>1. Let’s make a simple search</h2>
+
+      <Card.Root>
+        <Card.Content>
+          <Card.Highlight secondary>Next available</Card.Highlight>
+          <Card.Select
+            onSelect={(option) => setSort(option as any)}
+            defaultValue={sort}
+            options={[
+              { value: 'total_amount', label: 'cheapest flight' },
+              { value: 'total_duration', label: 'shortest flight' },
+            ]}
+          />
+          <Card.Text color="light">from</Card.Text>
+          <Card.Select
+            onSelect={setOrigin}
+            defaultValue={origin}
+            options={[
+              { value: 'JFK', label: 'JFK' },
+              { value: 'MIA', label: 'MIA' },
+              { value: 'LAX', label: 'LAX' },
+              { value: 'SFO', label: 'SFO' },
+            ]}
+          />
+          <Card.Text color="light">to</Card.Text>
+          <Card.Select
+            onSelect={setDestination}
+            defaultValue={destination}
+            options={[
+              { value: 'LHR', label: 'LHR' },
+              { value: 'CDG', label: 'CDG' },
+              { value: 'FRA', label: 'FRA' },
+              { value: 'MAD', label: 'MAD' },
+            ]}
+          />
+        </Card.Content>
+
+        <Card.Button
+          className="card__btn"
+          onClick={fetchCheapestOffer}
+          disabled={isFetching}
+        >
+          {isFetching ? 'Searching…' : 'Search'}
+        </Card.Button>
+      </Card.Root>
+    </Fragment>
+  )
+}
+
+interface BookingCardProps {
+  offer: Offer
+  onSuccess: (order: Order) => void
+}
+
+const BookingCard: FC<BookingCardProps> = ({ offer, onSuccess }) => {
+  const [isFetching, setIsFetching] = useState(false)
+  const [_errors, setErrors] = useState<ApiResponseError[]>([])
```

Comment:
probably doesn't really matter, but do we need to `setErrors` if we aren't going to use the errors? 😬